### PR TITLE
fix: dbconnections

### PR DESCRIPTION
**Pull request by markusahlstrand**
Migrate dbconnections to openapi

### CODE REVIEW

File: src/routes/tsx/routes.tsx [hunk 1]
```diff
@@ -165,6 +165,17 @@ export const login = new OpenAPIHono<{ Bindings: Env }>()
             description: "The state parameter from the authorization request",
           }),
         }),
+        body: {
+          content: {
```
Comment (markusahlstrand) [inline src/routes/tsx/routes.tsx:169]: I'm not 100% sure this is correct? So, we are posting form fields from the client right?

File: src/routes/tsx/routes.tsx [hunk 2]
```diff
@@ -206,54 +221,6 @@ export const login = new OpenAPIHono<{ Bindings: Env }>()
         tenantNameInVendorStyles,
       );
 
-      if (
```
Comment (markusahlstrand) [inline src/routes/tsx/routes.tsx:209]: We only support form fields now, so we can remove a lot of stuff. The 400's are handled by the framework

File: test/integration/flows/code-flow.spec.ts [hunk 3]
```diff
@@ -1377,22 +1377,14 @@ describe("code-flow", () => {
       // Now start password sign up with same code-user@example.com email
       // I'm seeing if this affects the code user with the same email address
       // -----------------
-      const typesDoNotWorkWithThisSetup___PARAMS = {
+      const createUserResponse = await loginClient.dbconnections.signup.$post({
```
Comment (markusahlstrand) [inline test/integration/flows/code-flow.spec.ts:1380]: This actually works now when the types are correct.

File: scripts/init-sqlite.ts [hunk 4]
```diff
@@ -51,6 +51,6 @@ const data = createAdapters(db);
 
   await data.passwords.create(tenant.id, {
     user_id: user.id,
-    password: "admin",
+    password: "Password1!",
```
Comment (danielgent) [inline scripts/init-sqlite.ts:54]: :smile:

File: src/routes/tsx/routes.tsx (hunk 1 above)
Comment (danielgent) [reply src/routes/tsx/routes.tsx:165]: I have no idea! I have two PRs now which are stuck due to this

https://github.com/sesamyab/auth/pull/646 & https://github.com/sesamyab/auth/pull/670

Seems really weird, and the docs I've found just show simple GET requests

I figure I've done enough testing now so I should actually look into this :see_no_evil:

File: src/routes/tsx/routes.tsx (hunk 2 above)
Comment (danielgent) [reply src/routes/tsx/routes.tsx:205]: Much better

File: test/integration/flows/__image_snapshots__/code-flow-spec-ts-test-integration-flows-code-flow-spec-ts-code-flow-edge-cases-should-login-correctly-for-a-code-account-linked-to-another-account-with-a-different-email-when-a-password-account-has-been-registered-but-not-verified-1-snap.png
Comment (danielgent) [inline test/integration/flows/__image_snapshots__/code-flow-spec-ts-test-integration-flows-code-flow-spec-ts-code-flow-edge-cases-should-login-correctly-for-a-code-account-linked-to-another-account-with-a-different-email-when-a-password-account-has-been-registered-but-not-verified-1-snap.png:1]: why are there any snapshot changes on this PR @markusahlstrand ?

I don't really like this at all. 

Can you reset the last two commits on this PR and then rebase of main and show me the failure?

Once we start committing snapshots without knowing why everything goes wrong.  

I don't see any extra assertions and the whole idea is that they *do not change*

**Comment by danielgent** [comment]
I don't intend to fix this PR @markusahlstrand I'm just checking there's nothing else missing

**Comment by danielgent** [comment]
All done!
